include/meta/meta.hpp

```diff
@@ -3137,21 +3136,20 @@ namespace meta
             struct impl
             {
             };
-            template <typename... Yes, typename... No, typename A>
+
+            template <template<typename...> class L, typename Fn, typename... As>
 #ifdef META_CONCEPT
-            requires Integral<invoke<Fn, A>>
-            struct impl<pair<list<Yes...>, list<No...>>, A>
+            requires (Integral<invoke<Fn, As>> && ...)
+            struct impl<L<As...>, Fn>
 #else
-            struct impl<pair<list<Yes...>, list<No...>>, A,
-                        void_<bool_<invoke<Fn, A>::type::value>>>
+            struct impl<L<As...>, Fn, void_<bool_<invoke<Fn, As>::type::value>...>>
 #endif
             {
-                using type = if_<invoke<Fn, A>, pair<list<Yes..., A>, list<No...>>,
-                                    pair<list<Yes...>, list<No..., A>>>;
+              using type = pair<filter<L<As...>, Fn>, filter<L<As...>, not_fn<quote<Fn::template invoke>>>>;
```

Comment:
Why `not_fn<quote<Fn::template invoke>>` instead of just `not_fn<Fn>` ?

Reply:
`not_fn<Fn>` is the better syntax; I don't remember why I wrote that more verbosely than is necessary. I'll recommit with the more concise syntax.

Reply:
Committed, fully built in my Appveyor/Travis accounts, build task 166.12 failed during libc++ installation. A restart of that task should pass.